frontend/app/components/input-file.tsx

```diff
@@ -0,0 +1,94 @@
+import { InputError } from '~/components/input-error';
+import { InputHelp } from '~/components/input-help';
+import { InputLabel } from '~/components/input-label';
+import { cn } from '~/utils/tailwind-utils';
+
+const inputBaseClassName =
+  'w-full sm:w-auto block rounded-sm border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300 focus:bg-neutral-300';
+const inputDisabledClassName =
+  'disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70';
+const inputErrorClassName = 'border-red-500 focus:border-red-500 focus:ring-red-500';
+
+export interface InputFileProps {
```

Comment:
Should this maybe extend `ComponentProps<'input'>` so callers can add additional `<input>` properties when it makes sense?